homeassistant/components/satel_integra/alarm_control_panel.py

```diff
@@ -68,16 +116,32 @@ def state(self):
 
     async def async_alarm_disarm(self, code=None):
         """Send disarm command."""
-        if code:
-            await self.hass.data[DATA_SATEL].disarm(code)
+        if not code:
+            _LOGGER.debug("Code was null")
+            return
+
+        clear_alarm_necessary = (self._state == STATE_ALARM_TRIGGERED)
+
+        _LOGGER.debug("clear_alarm_necessary: %s", clear_alarm_necessary)
+
+        await self.hass.data[DATA_SATEL].disarm(code)
+
+        if clear_alarm_necessary:
+            # Wait 1s before clearing the alarm
+            await asyncio.sleep(1)
+            await self.hass.data[DATA_SATEL].clear_alarm(code)
 
     async def async_alarm_arm_away(self, code=None):
         """Send arm away command."""
+        _LOGGER.debug("Arming away...")
```

Comment:
We don't use triple periods below. Please keep logging messages consistent in style.

Reply:
Will get fixed next push.